src/core/votetime.rs

```diff
@@ -35,39 +37,46 @@ pub struct TimeoutInfo {
 pub struct WaitTimer {
     timer_seter: Receiver<TimeoutInfo>,
     timer_notify: Sender<TimeoutInfo>,
-    thpool: ThreadPool,
+    // thpool: ThreadPool,
 }
 
 //unsafe impl ::std::marker::Sync for WaitTimer {}
 
 impl WaitTimer {
     pub fn new(ts: Sender<TimeoutInfo>, rs: Receiver<TimeoutInfo>) -> WaitTimer {
-        let pool = ThreadPool::new(THREAD_POOL_NUM);
+        // let pool = ThreadPool::new(THREAD_POOL_NUM);
         WaitTimer {
             timer_notify: ts,
             timer_seter: rs,
-            thpool: pool,
+            // thpool: pool,
         }
     }
 
     pub fn start(&self) {
         let innersetter = &self.timer_seter;
-        let zero_time = ::std::time::Duration::new(0, 0);
+        let mut timer_heap = min_max_heap::MinMaxHeap::new();
+
         loop {
-            select! {
-                settime = innersetter.recv() =>  {
-                    let oksettime = settime.unwrap();
-                    let notify = self.timer_notify.clone();
+            // take the peek of the min-heap-timer sub now as the sleep time otherwise set timeout as 0
+            let timeout = if !timer_heap.is_empty() {
+                timer_heap.peek_min().cloned().unwrap() - Instant::now()
+            } else {
+                Duration::from_millis(0)
```

Comment:
here should set a long duraton eg. 100s  or sync wait

Reply:
yes